#!/usr/bin/env python
PACKAGE = "mycontrol"
# 1.导包
from dynamic_reconfigure.parameter_generator_catkin import *

# 2.创建生成器
gen = ParameterGenerator()

# 3.向生成器添加若干参数
#add(name, paramtype, level, description, default=None, min=None, max=None, edit_method="")
gen.add("angle_pd",double_t,0,"直线与弯道角度D分辨系数，小于该参数使用直线PD参数，大于该参数使用弯道PD参数",0.25,0,3)
gen.add("angle_protect",double_t,0,"角度误差滤波参数，角度误差大于该参数将进行过滤，不作为控制参考",1,0,10)
gen.add("line_protect",double_t,0,"位置误差滤波参数，位置误差大于该参数将进行过滤，不作为控制参考",1,0,10)
gen.add("brake",double_t,0,"预警刹车参数，当角度误差变化大于该阈值时触发刹车",4,0,20)
gen.add("P_line",int_t,0,"位置y误差P系数",87,0,200)
gen.add("D_line",int_t,0,"位置y误差D系数",40,0,200)
gen.add("P_angle",int_t,0,"角度theta误差P系数",20,0,200)
gen.add("D_angle",int_t,0,"角度theta误差D系数",20,0,200)


# 4.生成中间文件并退出
exit(gen.generate(PACKAGE,"mycontrol","dr"))